File: app/src/main/java/desafio/com/br/github/ui/BaseActivity.java
package desafio.com.br.github.ui;

import android.app.*;
import android.support.annotation.*;

import butterknife.*;

/**
 * Created by rafael on 25/01/18.
 */

public abstract class BaseActivity extends Activity {

    @Override
    public void setContentView(@LayoutRes int layoutResID) {
        super.setContentView(layoutResID);
        ButterKnife.bind(this);
    }


}

File: Nexaas/common/src/main/java/br/com/nexaas/common/ui/base/BaseActivity.kt
package br.com.nexaas.common.ui.base

import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil.setContentView
import androidx.databinding.ViewDataBinding

abstract class BaseActivity<T : ViewDataBinding> : AppCompatActivity() {

    @LayoutRes
    abstract fun getLayoutRes(): Int

    abstract fun initView(savedInstanceState: Bundle?)

    protected val binding: T by lazy {
        setContentView<T>(this, getLayoutRes())
            .apply {
                lifecycleOwner = this@BaseActivity
            }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initView(savedInstanceState = savedInstanceState)
    }
}